src/ERC3009.sol

```diff
@@ -74,85 +213,132 @@ abstract contract ERC3009 is IERC3009, StatefulERC712 {
         bytes32 r_,
         bytes32 s_
     ) external {
-        if (msg.sender != to_) revert CallerMustBePayee(msg.sender, to_);
-
-        _transferWithAuthorization(
-            RECEIVE_WITH_AUTHORIZATION_TYPEHASH,
+        _revertIfInvalidSignature(
             from_,
-            to_,
-            value_,
-            validAfter_,
-            validBefore_,
-            nonce_,
-            v_,
-            r_,
-            s_
+            _getDigest(
+                keccak256(
+                    abi.encode(
+                        RECEIVE_WITH_AUTHORIZATION_TYPEHASH,
+                        from_,
+                        to_,
+                        value_,
+                        validAfter_,
+                        validBefore_,
+                        nonce_
+                    )
+                )
+            ),
+            abi.encodePacked(r_, s_, v_)
         );
+
+        _receiveWithAuthorization(from_, to_, value_, validAfter_, validBefore_, nonce_);
     }
 
     /// @inheritdoc IERC3009
-    function cancelAuthorization(address authorizer_, bytes32 nonce_, uint8 v_, bytes32 r_, bytes32 s_) external {
-        if (authorizationState[authorizer_][nonce_]) revert AuthorizationAlreadyUsed(authorizer_, nonce_);
+    function cancelAuthorization(address authorizer_, bytes32 nonce_, bytes memory signature_) external {
+        _revertIfInvalidSignature(
+            authorizer_,
+            _getDigest(keccak256(abi.encode(CANCEL_AUTHORIZATION_TYPEHASH, authorizer_, nonce_))),
+            signature_
+        );
 
+        _cancelAuthorization(authorizer_, nonce_);
+    }
+
+    /// @inheritdoc IERC3009
+    function cancelAuthorization(address authorizer_, bytes32 nonce_, bytes32 r_, bytes32 vs_) external {
         _revertIfInvalidSignature(
             authorizer_,
             _getDigest(keccak256(abi.encode(CANCEL_AUTHORIZATION_TYPEHASH, authorizer_, nonce_))),
-            v_,
-            r_,
-            s_
+            abi.encodePacked(r_, vs_)
```

Comment:
And here.

Reply:
Fixed in this commit: https://github.com/MZero-Labs/common/pull/8/commits/b63a1861b99ff78cc8356262b8076badf40154a4